scipy/signal/tests/test_signaltools.py

```diff
@@ -608,35 +762,38 @@ def test_lfiltic_bad_zi(self):
 
 
 class TestLinearFilterFloat32(_TestLinearFilter):
-    dt = np.float32
+    dtype = np.dtype('f')
 
 
 class TestLinearFilterFloat64(_TestLinearFilter):
-    dt = np.float64
+    dtype = np.dtype('d')
 
 
 class TestLinearFilterFloatExtended(_TestLinearFilter):
-    dt = np.longdouble
+    dtype = np.dtype('g')
 
 
 class TestLinearFilterComplex64(_TestLinearFilter):
-    dt = np.complex64
+    dtype = np.dtype('F')
 
 
 class TestLinearFilterComplex128(_TestLinearFilter):
-    dt = np.complex128
-
+    dtype = np.dtype('D')
 
-class TestLinearFilterComplexxxiExtended28(_TestLinearFilter):
-    dt = np.longcomplex
 
+class TestLinearFilterComplexExtended(_TestLinearFilter):
+    dtype = np.dtype('G')
 
 class TestLinearFilterDecimal(_TestLinearFilter):
-    dt = np.dtype(Decimal)
+    dtype = np.dtype('O')
+
+    def type(self, x):
+        return Decimal(str(x))
 
 
 class TestLinearFilterObject(_TestLinearFilter):
-    dt = np.object_
+    dtype = np.dtype('O')
+    type = float
```

Comment:
Yes.  This requires that you specify the type of object to use in the object arrays by providing either a callable type, as here , or a method, as in the decimal test.  This float object version is probably unnecessary if the the decimal tests pass.  I left it in since some else though it was worthwhile before and it doesn't add much overhead.